src/Middleware/HostFiltering/src/HostFilteringMiddleware.cs

```diff
@@ -80,6 +81,7 @@ private IList<StringSegment> EnsureConfigured()
         {
             if (_allowAnyNonEmptyHost == true || _allowedHosts?.Count > 0)
             {
+                Debug.Assert(_allowedHosts != null);
```

Comment:
💡 If you create a property `AllowAnyNonEmptyHost`, you can apply the `MemberNotNullWhenAttribute` to it to avoid needing this assertion. Another option is to make `_allowedHosts` never be null (I believe you can assign `Array.Empty<StringSegment>()` to it for the default case without changing the behavior).

Reply:
Hmm, wouldn't I have to move the assertion in to the `AllowAnyNonEmptyHost` property at that point? Perhaps I'm missing how we would tie to together.